node/src/components/block_validator/state.rs

```diff
@@ -120,14 +124,16 @@ impl BlockValidationState {
         responder: Responder<bool>,
         chainspec: &Chainspec,
     ) -> (Self, Option<Responder<bool>>) {
-        let deploy_count = block.deploys().len() + block.transfers().len();
-        if deploy_count == 0 {
+        let transaction_count = block.non_transfer_transactions().len() + block.transfers().len();
+        if transaction_count == 0 {
             let state = BlockValidationState::Valid(block.timestamp());
             return (state, Some(responder));
         }
 
-        if block.deploys().len() > chainspec.transaction_config.block_max_standard_count as usize {
-            warn!("too many non-transfer deploys");
+        if block.non_transfer_transactions().len()
+            > chainspec.transaction_config.block_max_standard_count as usize
```

Comment:
Shouldn't we be checking each of the three categories here rather than grouping standard, installer/upgrader and staking all together?

Reply:
Yes, we should. Updated block validator state is available in 9b80304356f3bebccfb095308af1fde508cb8324